src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -474,29 +460,29 @@ static void rr_handover(grpc_exec_ctx *exec_ctx, glb_lb_policy *glb_policy,
     glb_policy->pending_picks = pp->next;
     GRPC_LB_POLICY_REF(glb_policy->rr_policy, "rr_handover_pending_pick");
     pp->wrapped_on_complete_arg.rr_policy = glb_policy->rr_policy;
+    pp->wrapped_on_complete_arg.free_when_done = pp;
```

Comment:
heh, I was making this change in parallel before reading your comment, same thoughts. Done.